core/data/src/main/java/com/google/samples/apps/nowinandroid/core/data/repository/DefaultSearchContentsRepository.kt

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2023 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.samples.apps.nowinandroid.core.data.repository
+
+import com.google.samples.apps.nowinandroid.core.database.dao.NewsResourceDao
+import com.google.samples.apps.nowinandroid.core.database.dao.NewsResourceFtsDao
+import com.google.samples.apps.nowinandroid.core.database.dao.TopicDao
+import com.google.samples.apps.nowinandroid.core.database.dao.TopicFtsDao
+import com.google.samples.apps.nowinandroid.core.database.model.asExternalModel
+import com.google.samples.apps.nowinandroid.core.database.model.asFtsEntity
+import com.google.samples.apps.nowinandroid.core.network.Dispatcher
+import com.google.samples.apps.nowinandroid.core.network.NiaDispatchers.IO
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.combine
+import kotlinx.coroutines.flow.distinctUntilChanged
+import kotlinx.coroutines.flow.flatMapLatest
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.flow.mapLatest
+import kotlinx.coroutines.withContext
+import javax.inject.Inject
+
+class DefaultSearchContentsRepository @Inject constructor(
+    private val newsResourceDao: NewsResourceDao,
+    private val newsResourceFtsDao: NewsResourceFtsDao,
+    private val topicDao: TopicDao,
+    private val topicFtsDao: TopicFtsDao,
+    @Dispatcher(IO) private val ioDispatcher: CoroutineDispatcher,
+) : SearchContentsRepository {
+
+    override suspend fun populateFtsData() {
+        withContext(ioDispatcher) {
+            newsResourceFtsDao.insertAll(
+                newsResourceDao.getOneOffNewsResources().map { it.asFtsEntity() },
+            )
+            topicFtsDao.insertAll(topicDao.getOneOffTopicEntities().map { it.asFtsEntity() })
+        }
+    }
+
+    override fun searchContents(searchQuery: String): Flow<SearchResult> {
+        // Surround the query by asterisks to match the query when it's in the middle of
+        // a word
+        val newsResourceIds = newsResourceFtsDao.searchAllNewsResources("*$searchQuery*")
+        val topicIds = topicFtsDao.searchAllTopics("*$searchQuery*")
+
+        val newsResourcesFlow = newsResourceIds
+            .mapLatest { it.toSet() }
```

Comment:
```suggestion
            .map { it.toSet() }
```
map is enough, since `toSet()` is not suspend function

Reply:
It still makes sense to use `mapLatest` with non-suspend function on a flow with high emission rate.

Reply:
I'll keep mapLatest